release/models/nat/openconfig-nat.yang

```diff
@@ -0,0 +1,1008 @@
+        without traffic.";
+    }
+
+    leaf overload {
+      type boolean;
+      description
+        "Enable port overloading (PAT) allowing multiple internal
+        addresses to share the same external address with
+        different ports.";
+    }
+
+    leaf log-translations {
+      type boolean;
+      description
+        "Enable logging of NAT translations.";
+    }
+
+    leaf max-translations-per-address {
+      type uint32;
+      description
+        "Maximum number of concurrent translations per external
+        address.";
+    }
+  }
+
+  grouping nat-pool-state {
+    description
+      "Operational state data for NAT address pools.";
+
+    container counters {
+      description
+        "Counters for NAT pool operations.";
+
+      leaf allocated-addresses {
+        type yang:counter32;
+        description
+          "Number of addresses currently allocated from this pool.";
+      }
+
+      leaf available-addresses {
+        type yang:counter32;
+        description
+          "Number of addresses available in this pool.";
+      }
+
+      leaf active-translations {
+        type yang:counter64;
+        description
+          "Number of currently active translations using this pool.";
+      }
+
+      leaf acl-hit-count {
+        type yang:counter64;
+        description
+          "Number of packets matching the ACL set.";
+      }
+    }
+  }
+
+  grouping nat-mapping-config {
+    description
+      "Configuration parameters for static NAT mappings.";
+
+    leaf name {
+      type string;
+      description
+        "Unique name for the static mapping.";
+    }
+
+    leaf type {
+      type identityref {
+        base NAT_TYPE;
+      }
+      description
+        "Type of static mapping: source or destination.";
+    }
+
+    leaf internal-address {
+      type inet:ip-address;
+      description
+        "Internal (source) IP address. Must match the address
+        family configured for the NAT instance.";
+    }
+
+    leaf external-address {
+      type inet:ip-address;
+      description
+        "External (destination) IP address. Must match the address
+        family configured for the NAT instance.";
+    }
+
+    leaf internal-port {
+      type inet:port-number;
+      description
+        "Internal port number for NAPT mappings.";
+    }
+
+    leaf external-port {
+      type inet:port-number;
+      description
+        "External port number for NAPT mappings.";
+    }
+
+    leaf protocol {
+      type protocol-type;
+      description
+        "Protocol for the mapping.";
+    }
+
+    leaf acl-set {
+      type leafref {
+        path "/oc-acl:acl/oc-acl:acl-sets/oc-acl:acl-set/" +
+             "oc-acl:config/oc-acl:name";
+      }
+      description
+        "Reference to ACL set for this static mapping.";
+    }
+
+    leaf max-translations {
+      type uint64;
+      description
+        "Maximum number of concurrent translations.";
+    }
+
+    leaf enabled {
```

Comment:
What happens when this is disabled _after_ a translation has already happened? Does the existing translation continue to remain active, or are the existing active static mappings cleared?

Reply:
I'm not sure. Probably the same behavior as removing that map config entry altogether. Do you have a preference to include in the model?

Reply:
I'm removing all `enabled` leaves to avoid ambiguity.